python/setup.py

```diff
@@ -4,40 +4,7 @@
 from skbuild import setup
 
 setup(
-    name="rmm",
-    version="23.04.00",
-    description="rmm - RAPIDS Memory Manager",
-    url="https://github.com/rapidsai/rmm",
-    author="NVIDIA Corporation",
-    license="Apache 2.0",
-    classifiers=[
-        "Intended Audience :: Developers",
-        "Topic :: Database",
-        "Topic :: Scientific/Engineering",
-        "License :: OSI Approved :: Apache Software License",
-        "Programming Language :: Python",
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
-        "Programming Language :: Python :: 3.10",
-    ],
-    # Include the separately-compiled shared library
-    extras_require={"test": ["pytest"]},
+    license_files=["LICENSE"],
     packages=find_packages(include=["rmm", "rmm.*"]),
```

Comment:
It may be possible to move this to `pyproject.toml` too using [this feature]( https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#custom-discovery ). Though it is marked as beta

Reply:
Indeed, I was planning to revert the last couple of changes from your PR where you moved things back to setup.py because of features that were in beta. I'm going to test them out and use them if they work.

Reply:
OK, there seem to be some issues with this change. It results in the compiled shared libraries being placed into the `$pkg.data` directory instead of the `$pkg` directory. I suspect that this is a result of some incompatibility with scikit-build, but it may also be a setuptools bug since the feature is still in beta as you pointed out.

I may take a quick second look at this again next week, but if I don't find an easy fix I'm going to punt on this particular change for this PR. The ultimate goal of this migration for me is to move rmm (and RAPIDS) onto scikit-build-core, and if this problem is indeed the result of a scikit-build incompatibility it might just go away when using scikit-build-core. Leaving one bit of logic in setup.py should be OK for now.

Reply:
Yeah totally reasonable to punt on any of these. This one felt the most risky. So maybe we can turn it into an issue and revisit later

Reply:
I'll be revisiting fairly soon in all likelihood as I scope out `scikit-build-core` changes.

Reply:
OK upon further inspection I'm fairly certain that this is not a bug per se, but rather an expected limitation of using scikit-build that will be fixable when moving to scikit-build-core.